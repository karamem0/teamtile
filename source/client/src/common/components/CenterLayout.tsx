//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import Presenter from './CenterLayout.presenter';

interface CenterLayoutProps {
  children?: React.ReactNode
}

function CenterLayout(props: Readonly<CenterLayoutProps>) {

  const { children } = props;

  return (
    <Presenter>
      {children}
    </Presenter>
  );

}

export default CenterLayout;
