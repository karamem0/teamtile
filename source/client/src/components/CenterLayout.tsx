//
// Copyright (c) 2022 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import Presenter from './CenterLayout.presenter';

interface CenterLayoutProps {
  children?: React.ReactNode
}

function CenterLayout(props: CenterLayoutProps) {

  const { children } = props;

  return (
    <Presenter>
      {children}
    </Presenter>
  );

}

export default CenterLayout;
