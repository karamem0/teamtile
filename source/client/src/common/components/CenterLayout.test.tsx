//
// Copyright (c) 2021-2024 karamem0
//
// This software is released under the MIT License.
//
// https://github.com/karamem0/teamtile/blob/main/LICENSE
//

import React from 'react';

import { render } from '@testing-library/react';

import CenterLayout from './CenterLayout.presenter';

test('create shapshot', async () => {
  const params = {
    children: <React.Fragment />
  };
  const { asFragment } = render(<CenterLayout {...params} />);
  expect(asFragment()).toMatchSnapshot();
});
