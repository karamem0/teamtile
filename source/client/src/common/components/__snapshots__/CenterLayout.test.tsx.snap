// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create shapshot 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-justify-content: center;
  justify-content: center;
  min-height: calc(100vh - 0.5rem);
}

@media (width >= 600px) {
  .emotion-0 {
    min-height: calc(100vh - 2rem);
  }
}

<body>
  <div>
    <div
      class="emotion-0"
    />
  </div>
</body>
`;
